### PR TITLE
Auth-Feed-Explore-Profile

### PR DESCRIPTION
**Pull request by horsemaker**
- Implemented Auth Slice
- Implemented Redux Store
- Implemented Auth
- Add Pages

### CODE REVIEW

File: src/frontend/layouts/Main/Main.js [hunk 1]
```diff
@@ -11,10 +19,38 @@ const Main = () => {
...
+          element={
+            <RequireAuth>
+              <FeedPage />
+            </RequireAuth>
+          }
+        />
+        <Route
+          path="/explore"
+          element={
+            <RequireAuth>
```
Comment (prathmeshjagtap) [inline src/frontend/layouts/Main/Main.js:41]: you can make use of the outlet in react-router and wrap all requiredAuth into single route like this 
<Route element ={RequiredAuth}>
    <Route  path="/"element={<FeedPage />} />
</Route>

**Review by prathmeshjagtap — APPROVED** [review]
Code Is clean and Well structured
Functionalities are also working fine
Liked the UI it is Clean and Good
Good work overall

**Review by narharikale — APPROVED** [review]
clean and well structured code ✨
good to go

File: src/frontend/layouts/Main/Main.js (hunk 1 above)
Comment (horsemaker) [reply src/frontend/layouts/Main/Main.js:41]: Will surely look into this. Thank you.

**Pull request merged**